### PR TITLE
Part 2 of integration tests for GenericModelBinder

### PR DESCRIPTION
**Pull request by rynowak**
This includes a random sampling of combinations where a collection model
binder is binding another collection as an element type.

### CODE REVIEW

**Comment by rynowak** [comment]
@harshgMSFT will rebase on top of your changes after they are merged.

File: test/Microsoft.AspNet.Mvc.IntegrationTests/GenericModelBinderIntegrationTest.cs [hunk 1]
```diff
@@ -201,5 +201,368 @@ protected override Task<ModelBindingResult> BindModelCoreAsync(ModelBindingConte
...
+            // Arrange
+            var argumentBinder = ModelBindingTestHelper.GetArgumentBinder();
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "parameter",
+                ParameterType = typeof(Dictionary<string, int>[])
+            };
+
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.QueryString = new QueryString("?parameter[0][key0]=0");
```
Comment (harshgMSFT) [inline test/Microsoft.AspNet.Mvc.IntegrationTests/GenericModelBinderIntegrationTest.cs]: nit:suggest use a non default value

Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.IntegrationTests/GenericModelBinderIntegrationTest.cs]: sure, I'll update all of these

**Comment by harshgMSFT** [comment]
:shipit:

**Pull request closed without merging**